flink-table/flink-table-common/src/main/java/org/apache/flink/table/catalog/ObjectIdentifier.java

```diff
@@ -42,50 +45,88 @@
 @PublicEvolving
 public final class ObjectIdentifier implements Serializable {
 
-    private final String catalogName;
-
-    private final String databaseName;
+    static final String UNKNOWN = "<UNKNOWN>";
 
+    @Nullable private final String catalogName;
+    @Nullable private final String databaseName;
     private final String objectName;
 
     public static ObjectIdentifier of(String catalogName, String databaseName, String objectName) {
-        return new ObjectIdentifier(catalogName, databaseName, objectName);
+        if (Objects.equals(catalogName, UNKNOWN) || Objects.equals(databaseName, UNKNOWN)) {
+            throw new IllegalArgumentException(
+                    String.format("Catalog or database cannot be named '%s'", UNKNOWN));
+        }
+        return new ObjectIdentifier(
+                Preconditions.checkNotNull(catalogName, "Catalog name must not be null."),
+                Preconditions.checkNotNull(databaseName, "Database name must not be null."),
+                Preconditions.checkNotNull(objectName, "Object name must not be null."));
+    }
+
+    static ObjectIdentifier ofAnonymous(String objectName) {
```

Comment:
explain the reason for this method and why it should not be exposed in the future